vllm/distributed/device_communicators/shm_broadcast.py

```diff
@@ -196,7 +196,9 @@ def __init__(
             # see http://api.zeromq.org/3-3:zmq-setsockopt for more details
             self.local_socket.setsockopt(XPUB_VERBOSE, True)
             local_subscribe_port = get_open_port()
-            self.local_socket.bind(f"tcp://*:{local_subscribe_port}")
+            socket_addr = f"tcp://localhost:{local_subscribe_port}"
```

Comment:
cc @robertgshaw2-neuralmagic  I remember we met some `localhost` v.s. `127.0.0.1` problem before, zmq treats them differently.

Reply:
We could use `127.0.0.1` instead. See: https://stackoverflow.com/questions/6024003/why-doesnt-zeromq-work-on-localhost/8958414#8958414

Reply:
weird - i wonder why I didn't see any errors! I'll change it.

Reply:
I also did not have an error on my machine, but this definitely impacted users in the past

Reply:
Yeah, I wonder if it was a common enough mistake that they added special handling at some point. I even see it in their docs. https://github.com/zeromq/pyzmq/blob/0204cc420e0fe5e00e07fae77b61ccb9218061a4/docs/source/howto/eventloop.md?plain=1#L69